mergin/test/test_client.py

```diff
@@ -805,3 +806,136 @@ def test_logging(mc):
+        else:
+            shutil.copy(f_versioned, mp.fpath_meta(f_updated))
+        mc.download_file(project_dir, f_updated, f_versioned, version=f"v{ver}")
+        diff_file = mp.fpath_meta(f"changes_v{ver}.diff")
+        mp.geodiff.create_changeset(
+            mp.fpath_meta(f_updated), f_versioned, mp.fpath_meta(diff_file))
+
+        assert mp.geodiff.has_changes(diff_file)
+        assert mp.geodiff.changes_count(diff_file) == expected[ver]["nr_of_changes"]
+
+        changes_file = diff_file + ".changes"
+        mp.geodiff.list_changes_summary(diff_file, changes_file)
+        with open(changes_file, 'r') as f:
+            expected_changed_table = expected[ver]["table"]
+            expected_change = expected[ver]["change"]
+            expected_nr_of_changes = expected[ver]["nr_of_changes"]
+            changes = json.loads(f.read())["geodiff_summary"][0]
+            assert changes["table"] == expected_changed_table
+            assert expected_change in changes
+            assert changes[expected_change] == expected_nr_of_changes
+
+    # make sure there will be exception raised if a file doesn't exist in the version
+    with pytest.raises(ClientError, match=f"No {f_updated} exists at version v5"):
+        mc.download_file(project_dir, f_updated, f_versioned, version=f"v5")
+
+
+def test_download_diffs(mc):
+    """Test download diffs for a project file between specified project versions."""
+    test_project = 'test_download_diffs'
+    project = API_USER + '/' + test_project
+    project_dir = os.path.join(TMP_DIR, test_project)
+    download_dir = os.path.join(project_dir, "diffs")  # project for downloading files at various versions
+    f_updated = "base.gpkg"
+    diff_file = os.path.join(download_dir, f_updated + ".diff")
+
+    mp = create_versioned_project(mc, test_project, project_dir, f_updated, remove=False)
+
+    project_info = mc.project_info(project)
+    assert project_info["version"] == "v4"
+
+    # Download diffs of updated file between versions 1 and 2
+    mc.get_file_diff(project_dir, f_updated, diff_file, "v1", "v2")
+    assert os.path.exists(diff_file)
+    assert mp.geodiff.has_changes(diff_file)
+    assert mp.geodiff.changes_count(diff_file) == 1
+    changes_file = diff_file + ".changes1-2"
+    mp.geodiff.list_changes_summary(diff_file, changes_file)
+    with open(changes_file, 'r') as f:
+        changes = json.loads(f.read())["geodiff_summary"][0]
+        assert changes["insert"] == 1
+        assert changes["update"] == 0
+
+    # Download diffs of updated file between versions 2 and 4
+    mc.get_file_diff(project_dir, f_updated, diff_file, "v2", "v4")
+    changes_file = diff_file + ".changes2-4"
+    mp.geodiff.list_changes_summary(diff_file, changes_file)
+    with open(changes_file, 'r') as f:
+        changes = json.loads(f.read())["geodiff_summary"][0]
+        assert changes["insert"] == 0
+        assert changes["update"] == 1
```

Comment:
It would be good to have also a check for wrong input, e.g. to < from, or if asking for v4 to v5